tests/components/bmw_connected_drive/test_coordinator.py

```diff
@@ -13,116 +13,172 @@
+from . import BIMMER_CONNECTED_VEHICLE_PATCH, FIXTURE_CONFIG_ENTRY
 
 from tests.common import MockConfigEntry, async_fire_time_changed
 
+FIXTURE_ENTITY_STATES = {
+    "binary_sensor.m340i_xdrive_door_lock_state": "off",
+    "lock.m340i_xdrive_lock": "locked",
+    "lock.i3_rex_lock": "unlocked",
+    "number.ix_xdrive50_target_soc": "80",
+    "sensor.ix_xdrive50_rear_left_tire_pressure": "2.61",
+    "sensor.ix_xdrive50_rear_right_tire_pressure": "2.69",
+}
+
 
 @pytest.mark.usefixtures("bmw_fixture")
-async def test_update_success(hass: HomeAssistant) -> None:
-    """Test the reauth form."""
-    config_entry = MockConfigEntry(**FIXTURE_CONFIG_ENTRY)
+async def test_config_entry_update(
+    hass: HomeAssistant,
+    freezer: FrozenDateTimeFactory,
+) -> None:
+    """Test if the coordinator updates the refresh token in config entry."""
+    config_entry_fixure = deepcopy(FIXTURE_CONFIG_ENTRY)
+    config_entry_fixure["data"][CONF_REFRESH_TOKEN] = "old_token"
+    config_entry = MockConfigEntry(**config_entry_fixure)
     config_entry.add_to_hass(hass)
 
+    assert (
+        hass.config_entries.async_get_entry(config_entry.entry_id).data[
+            CONF_REFRESH_TOKEN
+        ]
+        == "old_token"
+    )
+
     await hass.config_entries.async_setup(config_entry.entry_id)
     await hass.async_block_till_done()
 
-    assert config_entry.runtime_data.last_update_success is True
+    assert (
+        hass.config_entries.async_get_entry(config_entry.entry_id).data[
+            CONF_REFRESH_TOKEN
+        ]
+        == "another_token_string"
+    )
 
 
 @pytest.mark.usefixtures("bmw_fixture")
 async def test_update_failed(
     hass: HomeAssistant,
     freezer: FrozenDateTimeFactory,
 ) -> None:
-    """Test the reauth form."""
+    """Test a failing API call."""
     config_entry = MockConfigEntry(**FIXTURE_CONFIG_ENTRY)
     config_entry.add_to_hass(hass)
 
     await hass.config_entries.async_setup(config_entry.entry_id)
     await hass.async_block_till_done()
 
-    coordinator = config_entry.runtime_data
-
-    assert coordinator.last_update_success is True
+    # Test if entities show data correctly
+    for entity_id, state in FIXTURE_ENTITY_STATES.items():
+        assert hass.states.get(entity_id).state == state
 
+    # On API error, entities should be unavailable
     freezer.tick(timedelta(minutes=5, seconds=1))
-
     with patch(
-        "bimmer_connected.account.MyBMWAccount.get_vehicles",
+        BIMMER_CONNECTED_VEHICLE_PATCH,
         side_effect=MyBMWAPIError("Test error"),
     ):
         async_fire_time_changed(hass)
         await hass.async_block_till_done()
 
-    assert coordinator.last_update_success is False
-    assert isinstance(coordinator.last_exception, UpdateFailed) is True
+    for entity_id in FIXTURE_ENTITY_STATES:
+        assert hass.states.get(entity_id).state == "unavailable"
+
+    # And should recover on next update
+    freezer.tick(timedelta(minutes=5, seconds=1))
```

Comment:
```suggestion
    freezer.tick(timedelta(minutes=5))
```
FYI if you want to wait 5 minutes you can just do 5, there's no need for an extra second margin :)

Reply:
Thanks, done that (and now retrieve the correct values from the component itself)